### PR TITLE
feat(upgrade): change to ethers

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- replaces Web3 with ethers.js
- upgrades all other packages
- removes useless scripts
- strengthens linting and formatting
- adapts prod and test code

## Why

- demand for ethers.js
- technical dept

## Refs

- [update rif-relay-sample-dapp](https://rsklabs.atlassian.net/browse/PP-357)
- https://github.com/rsksmart/rif-relay-server/pull/74
- https://github.com/rsksmart/rif-relay-client/pull/68

### CODE REVIEW

File: package.json
Comment (franciscotobar) [inline package.json:75]: The GENERATE_SOURCEMAP is set to false since since on react18 its showing multiple warnings, we need to identify how to resolve those warnings.

File: package.json
Comment (franciscotobar) [inline package.json:110]: Needs to be downgraded so the `react-materialize` modals can work without complaining about script.

File: src/index.tsx
Comment (franciscotobar) [inline src/index.tsx]: Needs to comment since the `react-materialize` modals cannot be opened with the current react version.

File: src/modals/Deploy.tsx
Comment (franciscotobar) [inline src/modals/Deploy.tsx]: The `RelayClient` does not use the `RequestConfig`, I think that we should remove it and use it in the `RelayProvider`

File: package.json
Comment (antomor) [reply package.json:75]: I'll create a task for this

File: package.json
Comment (antomor) [reply package.json:110]: I'll create a task to investigate more on this.

File: scripts/prepare
Comment (antomor) [inline scripts/prepare]: I don't think we need to build the dapp here.

File: src/App.tsx
Comment (antomor) [inline src/App.tsx]: We should use the URL configured

File: src/App.tsx
Comment (antomor) [inline src/App.tsx]: We could store process.env['REACT_APP_RIF_RELAY_CHAIN_ID'] in a const to be reused at line 121

File: src/Utils.ts
Comment (antomor) [inline src/Utils.ts]: If we don't need any more of this function, I'd delete it

File: src/index.tsx
Comment (antomor) [reply src/index.tsx]: We can create a task to address this issue specifically

File: src/index.tsx
Comment (antomor) [inline src/index.tsx]: I think we can remove it for now.

File: src/modals/Deploy.tsx
Comment (antomor) [inline src/modals/Deploy.tsx]: I know you haven't changed it, but I'd join those two log lines

File: src/modals/Deploy.tsx
Comment (antomor) [inline src/modals/Deploy.tsx]: Why do we always return true now instead of returning `receipt.status`?

File: src/modals/Execute.tsx
Comment (antomor) [inline src/modals/Execute.tsx]: Don't we need this function any more? If so, please let's delete it.

File: src/modals/Transfer.tsx
Comment (antomor) [inline src/modals/Transfer.tsx]: Shouldn't we retrieve this value either from the configuration or from the provider?

File: src/reportWebVitals.ts
Comment (antomor) [inline src/reportWebVitals.ts]: I think we could remove this part for now

**Review by antomor — CHANGES_REQUESTED** [review]
@franciscotobar Apart from some minor changes required, LGTM. Thank you for the amount of work required for this.

File: scripts/prepare
Comment (franciscotobar) [reply scripts/prepare]: updated

File: src/App.tsx
Comment (franciscotobar) [reply src/App.tsx]: updated

File: src/App.tsx
Comment (franciscotobar) [reply src/App.tsx]: updated

File: src/Utils.ts
Comment (franciscotobar) [reply src/Utils.ts]: updated

File: src/index.tsx
Comment (franciscotobar) [reply src/index.tsx]: updated

File: src/modals/Deploy.tsx
Comment (franciscotobar) [reply src/modals/Deploy.tsx]: updated

File: src/modals/Deploy.tsx
Comment (franciscotobar) [reply src/modals/Deploy.tsx]: updated

File: src/modals/Execute.tsx
Comment (franciscotobar) [reply src/modals/Execute.tsx]: We need it but its always returning false, Im not sure if its due to the fact of react update

File: src/modals/Transfer.tsx
Comment (franciscotobar) [reply src/modals/Transfer.tsx]: updated

File: src/reportWebVitals.ts
Comment (franciscotobar) [reply src/reportWebVitals.ts]: like remove the whole file?

File: package.json
Comment (antomor) [inline package.json]: These dependencies need to be updated once we merge the related PRs.

File: scripts/prepare
Comment (antomor) [inline scripts/prepare]: Is husky installed differently? If not, I think we should keep the prepare script.

File: package.json
Comment (franciscotobar) [reply package.json:75]: updated

File: src/modals/Execute.tsx
Comment (antomor) [reply src/modals/Execute.tsx]: I've just removed the commented code

File: src/reportWebVitals.ts
Comment (antomor) [reply src/reportWebVitals.ts]: Yes, exactly, I did it.

File: scripts/prepare
Comment (antomor) [reply scripts/prepare]: I've changed this to install husky in the prepare script section of package.json without having an additional file.

File: package.json
Comment (antomor) [reply package.json]: done

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**
